ckanext/geocat/tests/test_harvester.py

```diff
@@ -1,256 +1,103 @@
-import json
 import logging
 import os
 
-import ckan.plugins.toolkit as tk
 import ckantoolkit.tests.helpers as h
 import pytest
-import requests
-import requests_mock
-from ckan import model
-from ckan.common import config
 
-from ckanext.harvest import queue
+from ckanext.geocat.harvester import GeocatHarvester
+from ckanext.harvest.tests.lib import run_harvest
 
 log = logging.getLogger(__name__)
 
 __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
 
 MOCK_URL = "http://mock-geocat.ch"
 mock_record_url = "http://mock-geocat.ch/geonetwork/srv/eng/csw-BAKOM"
-mock_capabilities_url = (
-    "http://mock-geocat.ch/?version=2.0.2&request=GetCapabilities&service=CSW"
-)
-clear_solr_url = (
-    config.get("solr_url")
-    + "/update?stream.body=%3Cdelete%3E%3Cquery%3Ename:geocat-harvester%20OR%20organization:geocat_org%3C/query%3E%3C/delete%3E&commit=true"
-)
-
-
-@pytest.mark.ckan_config(
-    "ckan.plugins",
-    "ogdch ogdch_pkg harvest ckan_harvester geocat_harvester scheming_datasets fluent",
-)
-@pytest.mark.usefixtures("with_plugins", "clean_db", "clean_index")
-class TestGeocatHarvestFunctional:
-    @classmethod
-    @pytest.fixture(autouse=True, scope="class")
-    def setup_class(cls):
-        h.reset_db()
-
-        cls.gather_consumer = queue.get_gather_consumer()
-        cls.fetch_consumer = queue.get_fetch_consumer()
-
-    @pytest.fixture(autouse=True)
-    def setup(self):
-        queue.purge_queues()
-        requests.get(clear_solr_url)
-
-        self.user_dict = h.call_action(
-            "user_create",
-            name="test_admin",
-            email="testuser@example.com",
-            password="password",
-        )
-        user_obj = model.User.get(self.user_dict["name"])
-        user_obj.sysadmin = True
-        model.Session.add(user_obj)
-        model.Session.commit()
-
-        org_name = "test-org"
-        org_data_dict = {"name": org_name}
-
-        org_context = {
-            "model": model,
-            "session": model.Session,
-            "user": self.user_dict["name"],
-        }
-        org = tk.get_action("organization_create")(org_context, org_data_dict)
-        self.org_id = org["id"]
-
-        yield
-
-        h.reset_db()
-        queue.purge_queues()
-        requests.get(clear_solr_url)
-
-    @pytest.fixture(autouse=True)
-    def _get_or_create_harvest_source(self, setup, **kwargs):
```

Comment:
This method was marked as a pytest fixture with `autouse=True`, which meant that it was run before every test method. It isn't a fixture, but a helper method, so this just caused problems.